### ISSUE TITLE
mr.developer issue with zc.buildout 3.0.0a1

### ISSUE THREAD

**Issue by ale-rt** [buildout/buildout#515]
I tested zc.buildout with an existent virtualenv:
```
$ ./bin/pip3.8 install -IU zc.buildout==3.0.0a1
Looking in links: file:///home/ale/.pip-wheelhouse
Collecting zc.buildout==3.0.0a1
  Using cached zc.buildout-3.0.0a1-py2.py3-none-any.whl (158 kB)
Collecting pip
  Downloading pip-20.1.1-py2.py3-none-any.whl (1.5 MB)
     |████████████████████████████████| 1.5 MB 1.1 MB/s 
Collecting setuptools>=8.0
  Using cached setuptools-46.4.0-py3-none-any.whl (583 kB)
Collecting wheel
  Using cached wheel-0.34.2-py2.py3-none-any.whl (26 kB)
Installing collected packages: pip, setuptools, wheel, zc.buildout
Successfully installed pip-20.1.1 setuptools-46.4.0 wheel-0.34.2 zc.buildout-3.0.0a1
```
Then running buildout:
```
./bin/buildout
Setting socket time out to 60 seconds.
mr.developer: Queued ... for checkout.
Upgraded:
  pip version 19.2.3,
  wheel version 0.33.6;
Restarting.
Generated script '/home/ale/Code/plone/projects/pi52-py3/bin/buildout'.
Requires-Python support missing. 

Traceback (most recent call last):
  File "/home/ale/Code/plone/projects/pi52-py3/lib/python3.8/site-packages/zc/buildout/patches.py", line 69, in patch_PackageIndex
    from pip._internal.index.collector import HTMLPage
ModuleNotFoundError: No module named 'pip._internal.index.collector'; 'pip._internal.index' is not a package
Setting socket time out to 60 seconds.
mr.developer: Queued '...' for checkout.
While:
  Installing.
  Loading extensions.

An internal error occurred due to a bug in either zc.buildout or in a
recipe being used:
Traceback (most recent call last):
  File "/home/ale/Code/plone/projects/pi52-py3/lib/python3.8/site-packages/zc/buildout/buildout.py", line 2182, in main
    getattr(buildout, command)(args)
  File "/home/ale/Code/plone/projects/pi52-py3/lib/python3.8/site-packages/zc/buildout/buildout.py", line 660, in install
    self._load_extensions()
  File "/home/ale/Code/plone/projects/pi52-py3/lib/python3.8/site-packages/zc/buildout/buildout.py", line 1199, in _load_extensions
    ep.load()(self)
  File "/home/ale/.buildout/eggs/mr.developer-2.0.0-py3.8.egg/mr/developer/extension.py", line 293, in extension
    return Extension(buildout)()
  File "/home/ale/.buildout/eggs/mr.developer-2.0.0-py3.8.egg/mr/developer/extension.py", line 289, in __call__
    config.save()
  File "/home/ale/.buildout/eggs/mr.developer-2.0.0-py3.8.egg/mr/developer/common.py", line 622, in save
    options, settings, args = parse_buildout_args(self.buildout_args[1:])
  File "/home/ale/.buildout/eggs/mr.developer-2.0.0-py3.8.egg/mr/developer/common.py", line 452, in parse_buildout_args
    raise ValueError("Invalid option", '-' + op[0])
ValueError: ('Invalid option', '--')
```

Then I tried:
```
$ ./bin/pip3.8 install -IU pip==20.1.1  # current last version
Looking in links: file:///home/ale/.pip-wheelhouse
Collecting pip==20.1.1
  Using cached https://files.pythonhosted.org/packages/43/84/23ed6a1796480a6f1a2d38f2802901d078266bda38388954d01d3f2e821d/pip-20.1.1-py2.py3-none-any.whl
Installing collected packages: pip
Successfully installed pip-20.1.1
```

Then I rerun buildout once: at first it failed with the same `ValueError: ('Invalid option', '--')`.
The second time I rerun it worked.

**Comment by gotcha** [comment]
I think this is a bad interaction between mr.developer and the code I added to protect against endless loop when upgrading.

**Comment by gotcha** [comment]
I opened a PR to make `mr.developer` compatible with `buildout==3.0` : https://github.com/fschulze/mr.developer/pull/199

**Comment by ale-rt** [comment]
Wow! Thank you so much :)

**Comment by gotcha** [comment]
https://pypi.org/project/zc.buildout/3.0.0b1/ comes with a fix: #536.

@ale-rt, feel free to reopen if it is not the case.
